app/src/main/kotlin/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -1661,20 +1672,16 @@ class WebViewActivity :
                     },
                 ) { _, _ ->
                     lifecycleScope.launch {
-                        val url = serverManager.getServer(presenter.getActiveServer())?.let {
-                            val base = it.connection.getUrl(buttonRefreshesInternal) ?: return@let null
-                            base.toString().toUri()
-                                .buildUpon()
-                                .appendQueryParameter("external_auth", "1")
-                                .build()
-                                .toString()
-                        }
-                        failedConnection = if (buttonRefreshesInternal) "internal" else "external"
-                        if (url != null) {
-                            loadUrl(url = url, keepHistory = true, openInApp = true)
-                        } else {
-                            waitForConnection()
-                        }
+                        // TODO Once we do this it's going to keep the override until load is called again
+                        //  in onWindowFocusChanged that should happen right after the dialog is canceled.
+                        //  Then it might fail again if the user internal URL is always failing but the user is
+                        //  at home. (I guess)
```

Comment:
I think this needs to be handled before merging, but haven't tested it yet

Reply:
I felt a bit overwhelmed by this logic, what does it means for the rest of the app. Like the widget won't benefits from the switch to internal or external here. If for instance I'm on internal and it fails to load the webview and press go external it won't apply to the widget nor anything else than the webview.